app/controllers/synonyms/commodities/search_references_controller.rb

```diff
@@ -3,6 +3,26 @@ module Commodities
     class SearchReferencesController < Synonyms::SearchReferencesController
       private
 
+      # We special case the commodity with the overrides below because its id is formed of the composite keys of
+      # goods_nomenclature_item_id and the productline_suffix. Without this we would fail to be able to
+      # manage subheading commodities that have the same goods_nomenclature_item_id.
+
+      # Override used when destroying, updating or showing a search reference for a unique commodity.
+      def search_reference
+        @search_reference ||= search_reference_parent.search_references.find(params[:id]).tap do |synonym|
+          synonym.referenced_id = search_reference_parent.goods_nomenclature_item_id
+          synonym.productline_suffix = search_reference_parent.productline_suffix
+        end
+      end
+
+      # Override used when creating a search reference for a unique commodity.
+      def build_search_reference
+        search_reference_parent.search_references.build(search_reference_params.to_h).tap do |synonym|
+          synonym.referenced_id = search_reference_parent.goods_nomenclature_item_id
+          synonym.productline_suffix = search_reference_parent.productline_suffix
+        end
+      end
+
```

Comment:
Using `tap` makes things a bit hidden, 
maybe we can be more explicit without it, and removing duplication too.
I am not sure, maybe introducing a method `update_synonym` 🤔

Reply:
No strong feelings on this but this section makes it feel like something is missing - in theory you wouldn't expect to have to assign these manually

I'd expect `search_reference_parent.search_references.build` would assign both `reference_id` and `producline_suffix` - that being said without looking at Her's internals I've no idea if thats actually possible?

Can you do something like the following - this should then 'just work' with the composite keys I think?

```ruby
class SearchReference
  def reference_id=(id_or_composite_id)
    return super unless id.to_s.match?(%r{\d{10}-\d{2}})

    commodity_id, producline_suffix = split('-', 2)
    self.producline_suffix = producline)suffix
    super(commodity_id)
  end
end
```

Reply:
I'll have a play with this. I agree pushing the understanding of the id to the model feels appropriate

Reply:
Probably a bit nicer to test, too